pyproject.toml

```diff
@@ -1,3 +1,64 @@
+[build-system]
+requires = ["poetry>=0.12"]
+build-backend = "poetry.masonry.api"
+
+[tool.poetry]
+name = "myia"
+version = "0.0.0"
+description = "A linear algebra compiler for a subset of Python"
+authors = ["Mila development team"]
+license = "MIT"
+readme = "README.rst"
+repository = "https://github.com/mila-iqia/myia"
+
+[tool.poetry.dependencies]
+python = "~3.7"
+asttokens = "^2.0.3"
+colorama = "^0.4.3"
+numpy = "^1.16"
+opt_einsum = "^3.1.0"
+prettyprinter = "^0.18.0"
+antlr4-python3-runtime = { version = "4.7.2", optional = true }
+torch = { version = "1.5.0", optional = true }
+torchvision = { version = "0.6.0", optional = true }
+
+[tool.poetry.extras]
+# tvm version is managed by relay-extras.conda
+relay = ["antlr4-python3-runtime"]#, "tvm"]
+pytorch = ["torch", "torchvision"]
+
+# versions for these are managed by {gpu,cpu}-extras.conda
```

Comment:
Why? Can't they be specified here, and the channels listed in `tool.poetry2conda.dependencies`?

Reply:
Because there are no packages on pypi for them and poetry tries to fetch metadata for every package listed in the file, which fails if they are listed.

Reply:
Even if the extra isn't asked for? Well, that's awful.